mechanicalsoup/form.py

```diff
@@ -277,6 +277,44 @@ def set(self, name, value, force=False):
             return
         raise LinkNotFoundError("No valid element named " + name)
 
+    def __getitem__(self, name):
+        """Forward arguments to :func:`~Form.get` to get value for a given form field.
+        For example, :code:`name = form["name"]` calls :code:`form.get("name")`
+        """
+        return self.get(name)
+
+    def get(self, name):
+        """Get a ``value`` from input element identified by ``name``.
+        Currently it does support "input" type only.
+
+        Example:
+
+        .. code-block:: python
+            firstname = browser['firstname']
+
+        If the element is not found , raise a :class:`LinkNotFoundError`
+        exception.
+
+"""
+        inp = self.form.find("input", {"name": name})
+        if not inp:
+            raise LinkNotFoundError("No input field named " + name)
+        return inp["value"]
```

Comment:
It is suspicious that we have such complex logic in `set` and that `get` is so straightforward.

I think your code works only on text fields. For example, for radio buttons you need to look for the option having `checked="checked"`. This is more or less what `Browser.get_request_kwargs` does.

It's OK to deal only with text fields as a limitation, but the code should throw `NotImplementedError` for non-managed types of fields (so that we can later add proper management for them without breaking compatibility).

Reply:
It indeed does, as state in PR message it works only with 'input' type fields